contracts/eosio.system/src/finalizer_key.cpp

```diff
@@ -6,12 +6,13 @@ namespace eosiosystem {
 
    // Returns true if nodeos has transitioned to Savanna (last finalizer set not empty)
    bool system_contract::is_savanna_consensus() const {
```

Comment:
Instead of `is_savanna_consensus()`, you could have a `get_last_proposed_finalizers` which returns `std::vector<proposed_finalizer_key_t>`. If the returned vector is empty, then we know that we have not yet switched to Savanna.